components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml

```diff
@@ -4866,6 +4869,193 @@ document.observe("xwiki:dom:loaded", function() {
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>var PhenoTips = (function(PhenoTips) {
+  // Start PhenoTips augmentation
+  var widgets = PhenoTips.widgets = PhenoTips.widgets || {};
+
+  widgets.DurationValidator =  Class.create({
+    pieces : {
+      y : 'years',
+      m : 'months',
+      w : 'weeks',
+      d : 'days',
+      n : '([1-9][0-9]*)',
+      s : '\\s*'
+    },
+    initialize : function(element) {
+      if (!element) {
+        return;
+      }
+      var sep = this.pieces.s;
+      this.subgroups = {
+         y  : this._regexpifyDurationUnit('y', true),
+         m  : this._regexpifyDurationUnit('m', true),
+         w  : this._regexpifyDurationUnit('w', true),
+         d  : this._regexpifyDurationUnit('d', true)
+      };
+      this.regexp = new RegExp("^((" + this.subgroups.y + sep + this.subgroups.m + ")|" + this.subgroups.w + ")" + sep + this.subgroups.d + "$");
+      this.element = element;
+      var _this = this;
+      ['keyup', 'input'].each(function(ev) {
+        element.observe(ev, function(event) {
+          if (_this.regexp.match(element.value)) {
+            element.removeClassName('error');
+          } else {
+            element.addClassName('error');
+          }
+        });
+      });
+      element.observe('blur', function(event) {
+        element.value = _this.format(element.value);
+        Element.fire(element, 'duration:format');
+        element.title = _this.getValue(element.value);
+      });
+    },
+    _regexpifyDurationUnit : function(unit, makeOptional) {
+      return "(" + this.pieces.s + this.pieces.n + this.pieces.s + this._regexpifyWord(this.pieces[unit] || "") + this.pieces.s + ")" + (makeOptional ? '?' : '');
+    },
+    _regexpifyWord : function(word) {
+      return word.replace(/(.)/g, "$1?").replace("?", "");
+    },
+    match : function(text) {
+      return (text || "").match(this.regexp);
+    },
+    getValue : function(text) {
+      if ((text || "").match(this.regexp)) {
```

Comment:
can't you just use `match` defined above? There are other occurrences of this in the file.